src/NextGen/DonationForm/Blocks/DonationFormBlock/resources/app/templates/index.tsx

```diff
@@ -1,195 +1,91 @@
-        ? withWrapper(template[section][type], section, type, htmlTag)
-        : null;
-
-    let FilteredNode = applyFilters(`givewp/form/${section}/${type}`, Node);
-    FilteredNode = applyFilters(`givewp/form/${section}`, Node, type);
-
-    if (nodeIsFunctionalComponent(FilteredNode)) {
-        return FilteredNode as FC<NodeProps>;
-    } else {
-        throw new Error(`Invalid field type: ${type}`);
-    }
+import type {FC, ReactNode} from 'react';
+import {useMemo} from 'react';
+import getFormDesign from "@givewp/blocks/form/app/utilities/getFormDesign";
+
+/**
+ * Get the active template from the window
+ *
+ * @unreleased
+ */
+const template = getFormDesign();
+
+/**
+ * Get the NodeWrapper from active template
+ *
+ * @unreleased
+ */
+const NodeWrapper = template.layouts.wrapper;
+
+/**
+ * Find the names of nodeType and type based on the value of the template.
+ *
+ * @unreleased
+ */
+export function findTemplateKeys<S extends keyof typeof template, T extends keyof typeof template[S]>(
+    templateValue: ReactNode
+): {nodeType: S | null; type: T | null} {
+    let nodeType = null;
+    let type = null;
+
+    Object.entries(template).forEach((firstLevel) => {
+        if (typeof firstLevel[1] === 'object') {
+            const firstLevelKey = firstLevel[0];
+            const secondLevelKeys = Object.keys(firstLevel[1]);
+            const templateType = secondLevelKeys.find((key) => firstLevel[1][key] === templateValue);
+
+            if (templateType) {
+                type = templateType;
+                nodeType = firstLevelKey;
+            }
+        }
+    });
+
+    return {nodeType, type};
 }
 
-export function getFieldTemplate(type: string): FC<FieldProps> {
-    return getTemplate<FieldProps>(type, 'fields');
-}
-
-export function getElementTemplate(type: string): FC<ElementProps> {
-    return getTemplate<ElementProps>(type, 'elements');
-}
-
-export function getGroupTemplate(type: string): FC<GroupProps> {
-    return getTemplate<GroupProps>(type, 'groups');
-}
-
-export function getSectionTemplate(): FC<SectionProps> {
-    return getTemplate<SectionProps>('section', 'layouts', 'section');
-}
-
-export function getFormTemplate(): FC<FormProps> {
-    return getTemplate<FormProps>('form', 'layouts');
-}
-
-export function getFieldLabelTemplate(): FC<FieldLabelProps> {
-    return getTemplate('fieldLabel', 'layouts');
-}
-
-export function getFieldErrorTemplate(): FC<FieldErrorProps> {
-    return getTemplate('fieldError', 'layouts');
-}
-
-export function getHeaderTemplate(): FC<HeaderProps> {
-    return getTemplate('header', 'layouts');
-}
-
-export function getHeaderTitleTemplate(): FC<HeaderTitleProps> {
-    return getTemplate('headerTitle', 'layouts');
+/**
+ * This HOC will wrap a template component in our NodeWrapper and automatically figure out what nodeType and type to use as props.
+ *
+ * @unreleased
+ */
+export function withTemplateWrapper<TemplateProps>(
```

Comment:
I'm on the fence, but I'm curious what made you decide to keep the template wrappers in the form block code?

Reply:
TBH, I didn't put much thought about _where_ these should live - I was just working from this file as it was previously.  I will say the template wrapper is very specific to the form rendering process and needs to work with the final form design template.  This is not something we want to expose, so I think it can either stay here or move to the root of the domain.

Reply:
This made me realize that the entire _form block app_ has no more business being inside the block domain anymore - since all the block does is render an iframe now.  I moved the donation form app inside our root domain's resources folder e125aa47 which further isolates the donation form as a standalone application that does not depend on the block itself.